backend/internal/auth/infrastructure/authRepository.go

```diff
@@ -0,0 +1,49 @@
+package infrastructure
+
+import (
+	"database/sql"
+	"errors"
+
+	"github.com/unmsmfisi-socialapplication/social_app/internal/auth/application"
+	"github.com/unmsmfisi-socialapplication/social_app/internal/auth/domain"
+)
+
+type AuthSessionDBRepository struct {
+	db *sql.DB
+}
+
+func NewAuthSessionDBRepository(database *sql.DB) domain.AuthSessionRepository {
+	return &AuthSessionDBRepository{db: database}
+}
+
+func (a *AuthSessionDBRepository) CheckValidSession(username string) (bool, error) {
+	query := `SELECT logged FROM SOC_APP_auth_sessions WHERE user_name = $1 ORDER BY timestamp DESC LIMIT 1`
```

Comment:
In the worst case scenario regarding the management session, how many records per user could exist, as you are limiting the result to `1`.

Why are you using a relational database for saving sessions?
Have you considered working with [transient](https://www.quora.com/What-is-a-transient-database) databases for this kind of situation? 

Is it valid for you to add a filter like 

 ```suggestion
	query := `SELECT logged FROM SOC_APP_auth_sessions WHERE user_name = $1 and logged != true ORDER BY timestamp DESC LIMIT 1`
```

Reply:
@ekimike Thank you for the feedback. I have already addressed these questions and have been clarified to you during our last Google Meet session.